### PR TITLE
Add "Allowed Tags" to Parental Controls

### PR DESCRIPTION
**Pull request by Jpuc1143**
**Changes**
Parental Controls  can now block all the media that do not have a specified tag, similar to the already existing "Block items with tags" but with a whitelist instead. By default, as long as no tags are configured for the whitelist, no shows will be blocked.

**Issues**
Implements #928

### CODE REVIEW

**Comment by TalEdd** [comment]
Would love this change to be merged to main as soon as possible, it would be a great feature to have

**Comment by Raik0** [comment]
Hopefully this can be looked at sooner rather than later

**Review by daullmer — CHANGES_REQUESTED** [review]

**Comment by daullmer** [comment]
Adding the setting in the web client would also be nice :)

File: MediaBrowser.Controller/Entities/BaseItem.cs [hunk 1]
```diff
@@ -1607,6 +1607,11 @@ private bool IsVisibleViaTags(User user)
                 return false;
             }
 
+            if (user.GetPreference(PreferenceKind.AllowedTags).Any() && !user.GetPreference(PreferenceKind.AllowedTags).Any(i => Tags.Contains(i, StringComparison.OrdinalIgnoreCase)))
```
[the remark this answers is not in the document]
Comment (daullmer) [reply MediaBrowser.Controller/Entities/BaseItem.cs:1610]: Yeah that makes sense. Didn't think that removing that line would leave no way of disabling this feature.

**Review by daullmer — APPROVED** [review]

File: MediaBrowser.Controller/Entities/BaseItem.cs (hunk 1 above)
Comment (crobibero) [inline MediaBrowser.Controller/Entities/BaseItem.cs:1610]: Should get the preference once.

```suggestion
            var allowedTagsPreference = user.GetPreference(PreferenceKind.AllowedTags);
            if (allowedTagsPreference.Any() && !allowedTagsPreference.Any(i => Tags.Contains(i, StringComparison.OrdinalIgnoreCase)))
            {
                return false;
            }
```

**Review by crobibero — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Comment by Raik0** [comment]
Merged!!!

**Comment by 0x11DFE** [comment]
Would also be nice having a way to sort by custom parental rating since sometimes it fails to detect the rating e.g FR-TP (Tout Public) is not available and is sorted as if it was 18+.
Also i had a similar issue where some adult Horror movies were present in the kids profile maybe metadata wasn't able to fetch from DB correctly?

**Comment by adoreparler** [comment]
Is this working?

I am on version 10.9.0

Screen Recording 2024-04-17 at 9 14 59 AM

**Comment by crobibero** [comment]
@adoreparler Please create an issue for tracking instead of commenting on the PR.

**Pull request merged**